### ISSUE TITLE
RPM dependency issue with latest agent v9.3.0 

### ISSUE THREAD

**Issue by dciancio** [F5Networks/f5-openstack-agent#663]
#### Agent Version
f5-openstack-agent-9.3.0-1.el7.noarch.rpm

#### Operating System
RHEL 7.3

#### OpenStack Release
RHEL OSP v9 (mitaka)

#### Description
SDK RPM fails to install due to dependency issue.

[root@isft502 9.3.0]# rpm -qa | grep six
python-six-1.9.0-2.el7.noarch

[root@isft502 9.3.0]# rpm -ivh f5-icontrol-rest-1.3.0-1.el7.noarch.rpm
Preparing...                          ################################# [100%]
        package f5-icontrol-rest-1.3.0-1.noarch is already installed

[root@isft502 9.3.0]# rpm -ivh f5-sdk-2.3.1-1.el7.noarch.rpm
error: Failed dependencies:
        f5-icontrol-rest<2.0.0 is needed by f5-sdk-2.3.1-1.noarch
        f5-icontrol-rest>=1.3.0 is needed by f5-sdk-2.3.1-1.noarch
        six<2.0.0 is needed by f5-sdk-2.3.1-1.noarch
        six>=1.9.0 is needed by f5-sdk-2.3.1-1.noarch

Please fix RPM issue.  Thanks!

**Comment by dciancio** [comment]
The "latest" documentation link references a version of the SDK rpm (v2.3.1) which is broken.
Using the "9.3.0" doc link references v2.2.2 of the SDK rpm which installs fine.
You may want to correct your documentation links.

**Comment by pjbreaux** [comment]
@dciancio: we are producing a new f5-sdk package that you will be able to use. Stay tuned. Thanks very much.

**Comment by ssorenso** [comment]
@dciancio: If you need to get up and running right away, you can add the ```--nodeps``` option.  We see that you have the dependencies, as long as you still have python-six >= 1.9.0; <2.0.0 and the f5-icontrol-rest library still installed.

```
[root@5706a21df984 build]# rpm -i ./f5-sdk-2.2.2-1.el7.noarch.rpm
error: Failed dependencies:
	f5-icontrol-rest<2.0.0 is needed by f5-sdk-2.2.2-1.noarch
	f5-icontrol-rest>=1.3.0 is needed by f5-sdk-2.2.2-1.noarch
	python-six>=1.9.0 is needed by f5-sdk-2.2.2-1.noarch
	pytohn-six<2.0.0 is needed by f5-sdk-2.2.2-1.noarch
[root@5706a21df984 build]# ls
f5-sdk-2.2.2-1.el7.noarch.rpm
[root@5706a21df984 build]# rpm -i --nodeps ./f5-sdk-2.2.2-1.el7.noarch.rpm
[root@5706a21df984 build]#
```

As far as the issue, we do have automated tests that attempt to install the newly generated RPM in a docker instance that is running CentOS 7 natively (similar to the docker file that generates it).  This Docker file has the following settings:
```
# Dockerfile
FROM centos:7

RUN yum update -y && yum install git python-requests -y
RUN yum install -y python-six

COPY ./fetch_and_install_deps.py /

# ENTRYPOINT ["python", "-m", "pdb", "/fetch_and_install_deps.py", "/var/wdir"]
ENTRYPOINT ["python", "/fetch_and_install_deps.py", "/var/wdir"]
# by default, the following will be used for the last arg:
CMD ["/var/wdir/f5-sdk-dist/rpms/build/f5-sdk-*.el7.noarch.rpm"]
```

Just for your reference.  The script there finds the dependencies and installs them, then attempts the install on the RPM as a test.  Upon failure, it should stop our travis build as a failure.  That's the expectation, so now I'm trying to find out why that's not working.

**Comment by ssorenso** [comment]
So the RCA is in 2 parts:

1. The call to `rpm -i <package>` in the aforementioned script does not raise on a non-zero exit status
  * This would cause it to fail in Travis appropriately
2. The Requires: format that the RPM's spec file is expecting a different format
  * The format that exists is 'package>=version', but it should be 'package >= version, ' with the ', ' delimitation between requirements instead of just a space

As such, I've got a fix that I'm about to push up that will resolve this in the F5-SDK.

**Comment by dflanigan** [comment]
We need a new SDK release to fix and close this bug.

**Comment by pjbreaux** [comment]
This was fixed with the release of 2.3.2 in the sdk and with the closure of this sdk issue:

https://github.com/F5Networks/f5-common-python/issues/1116

**Comment by dflanigan** [comment]
Closing because bug was fixed with release of sdk.

From @pjbreaux : I don’t know if this was referenced in any PR. So we can just close it outright since the releases are out the door. Let me know if that sounds good to you.
